sw/device/tock/chips/opentitan_common/src/uart.rs

```diff
@@ -220,23 +110,29 @@ impl Uart<'a> {
     }
 
     pub fn handle_interrupt(&self) {
-        let regs = self.registers;
-        let intrs = regs.intr_state.extract();
-
-        if intrs.is_set(intr::tx_empty) {
-            self.disable_tx_interrupt();
-            self.tx_progress();
-        } else if intrs.is_set(intr::rx_watermark) {
-            self.disable_rx_interrupt();
-            // TODO: real RX processing
+        let interrupt = kDifUartInterruptTxWatermark;
+        match self.interrupt_state(interrupt) {
+            Ok(set) => {
+                if set {
+                    self.tx_buffer_process();
+                    self.interrupt_clear(interrupt); // TODO return this
```

Comment:
the TODO here is because of the interface of `handle_interrupt` right?

Reply:
Yes that is correct, pretty much all of the functions before this change, were "void". This is an open issue that is linked to #2139 - namely, how far do we wan't to propagate errors and what to do with them.